app/service/app_svc.py

```diff
@@ -82,9 +84,19 @@ async def run_scheduler(self):
         while True:
             interval = 60
             for s in await self.get_service('data_svc').locate('schedules'):
-                now = datetime.now(timezone.utc).time()
-                today_utc = datetime.now(timezone.utc).date()
-                diff = datetime.combine(today_utc, now) - datetime.combine(today_utc, s.schedule)
+                now = datetime.now(timezone.utc)
+                match = re.match(r'^(\d{2}):(\d{2}):\d{2}\.\d{6}$', s.schedule)
```

Comment:
Ideally this check should only happen once - that way we avoid unnecessary regex checks on schedules whose formats have already been validated. It looks like in https://github.com/mitre/caldera/pull/3026, the format is validated on schedule creation, so this check will eventually only be relevant for preexisting schedules that are restored from a saved state. For now, we can perform this check and conversion if `croniter.is_valid` returns `false` [below](https://github.com/mitre/caldera/pull/3025/files#diff-155075fcc9f0ddaccbd732abcfe1cb04a295f470a02f09b64ac6014b6e404783R94), and then send out the warning if the regex check fails as well, since by then we know that the schedule ID does not adhere to either format

Reply:
I've updated this section I let you check if it's okay for you ?

Reply:
great, thank you!